### ISSUE TITLE
Support hasMany relationship with embedded data

### ISSUE THREAD

**Issue by Tuizi** [BookingSync/ember-data-partial-model#26]
Does this case is supported by **ember-data-partial-model**.

Here is my sample model:

``` javascript
// model/form.js
export default PartialModel.extend({
    key: DS.attr('string'),
    extended: partial("extended", "form", {
        answers: DS.hasMany('answer', {async: false})
    })
});
```

``` javascript
// serializers/form.js
export default ApplicationSerializer.extend({
    attrs: {
        answers: {embedded: 'always'}
    }
});
```

```
// api/forms
{
   id: 1
   key: 2
}

// api forms/1
{
   id: 1
   key: 2
   answers: [{id: 1, text: "Hello"}]
}
```
1. First of all with that I have this warning: **There is no attribute or relationship with the name `answers` on `form`. Check your serializers attrs hash.** 
2. When I do `form.get("answers")` a call to the backend is made but I got this error: **You need to pass a model name to the store's modelFor method**

It looks like it fail because in the answer model there is not property `type`, is it normal? If I use "normal" model, without PartialModel, it's working as expected. 

Any idea?

**Comment by Azdaroth** [comment]
Hi @Tuizi,

I don't think we've ever tested partials with `embedded` option, so this could be indeed an issue.

What kind of serializer are you using? 

I'm wondering if that's really the best way to solve your problem, do you really need to put embeddable `hasMany` relationship in partial? Wouldn't it work kind of the same with association in top-level model with `async: true`?

**Comment by Tuizi** [comment]
I'm using this serializer:

``` javascript
import DS from 'ember-data';
import PartialModelRESTSerializer from 'ember-data-partial-model/mixins/rest-serializer';

export default DS.RESTSerializer.extend(PartialModelRESTSerializer);
```

I'm not sure to understand your point with `async:true` do you have an example? :)

**Comment by Azdaroth** [comment]
Well, it basically comes down to the question if embedding is really needed in such case? Wouldn't it work that way? Could be even better as it doesn't require to actually load the answers immediately.

```
export default PartialModel.extend({
    key: DS.attr('string'),
    extended: partial("extended", "form", {
        answers: DS.hasMany('answer')
    })
});
```

Let me know how it behaves, if there's still any issue I'll take a look at it this weekend.

**Comment by Tuizi** [comment]
I'm agree with you, maybe our REST API is not correct. Right know I want to test **ember-data-partial-model** and migrate our actual model with it. But we can't because we have a lot of embedded... maybe you are right.

But, as this feature is supported by ember-data, it could be a good improvement for ember-data-partial-model no?

**Comment by Azdaroth** [comment]
Yeah, it makes sense to support it in such case, I should have something ready by the end of this week.

**Comment by Tuizi** [comment]
Thank you @Azdaroth, meanwhile I review my rest api ;)

**Comment by Azdaroth** [comment]
Hi @Tuizi ,

can you check this PR our before new version is released? https://github.com/BookingSync/ember-data-partial-model/pull/27/files

I updated the Readme as well, so I hope it will be clear enough what changes need to be done. 

I took a bit more flexible approach and added support for extending partial serializers, so now you can apply any extensions and add any mixins for partial serialisers you want :).

BTW., the serializer for Answer model should albo be REST one, so that the formats are not mixed.

**Comment by Tuizi** [comment]
Hi! It's perfect... I read the **README.md** and it's exactly what we needed. 
Thank you very much

**Comment by Azdaroth** [comment]
@Tuizi Released 0.4.0 version, so you may want update the addon to the latest version to keep it clean :).
